sklearn/ensemble/gradient_boosting.py

```diff
@@ -1034,6 +1034,10 @@ def fit(self, X, y, sample_weight=None, monitor=None):
                                  % (self.n_estimators,
                                     self.estimators_.shape[0]))
             begin_at_stage = self.estimators_.shape[0]
+            # The requirements of _decision_function (called in two lines
+            # below) are more constrained than fit. It accepts only CSR
+            # matrices.
+            X = check_array(X, dtype=DTYPE, order="C", accept_sparse='csr')
```

Comment:
`order="C"` [doesn't seem to be used in `check_array`](https://github.com/scikit-learn/scikit-learn/blob/a24c8b46/sklearn/utils/validation.py#L433) for sparse arrays. 

That's an unrelated fix then? What happened previously if the cython code got dense fortran ordered arrays; the results were wrong?

Reply:
It is indeed needed since at decision_function will expect a C array at the moment of the prediction:

https://github.com/scikit-learn/scikit-learn/blob/a24c8b464d094d2c468a16ea9f8bf8d42d949f84/sklearn/ensemble/_gradient_boosting.pyx#L65

Reply:
OK, thanks, makes sense.